src/nisarqa/processing/insar/quiver_plots.py

```diff
@@ -352,26 +580,99 @@ def truncate_colormap(
     arrow_stride = int(max(np.shape(total_offset)) / params.arrow_density)
 
     # Only plot the arrows at the requested strides.
-    arrow_y = az_off[::arrow_stride, ::arrow_stride]
-    arrow_x = rg_off[::arrow_stride, ::arrow_stride]
+    az_offsets_at_arrow_tails = az_off[::arrow_stride, ::arrow_stride]
+    rg_offsets_at_arrow_tails = rg_off[::arrow_stride, ::arrow_stride]
 
-    x = np.linspace(0, arrow_x.shape[1] - 1, arrow_x.shape[1])
-    y = np.linspace(0, arrow_x.shape[0] - 1, arrow_x.shape[0])
+    num_arrows_along_x = az_offsets_at_arrow_tails.shape[1]
+    num_arrows_along_y = az_offsets_at_arrow_tails.shape[0]
+    x = np.linspace(0, num_arrows_along_x - 1, num_arrows_along_x)
+    y = np.linspace(0, num_arrows_along_y - 1, num_arrows_along_y)
     X, Y = np.meshgrid(x, y)
 
+    # 2D grid of the starting x pixel indices in the image arrays for each arrow
+    arrow_tails_x_indices = X * arrow_stride
+
+    # 2D grid of the starting y pixel indices in the image arrays for each arrow
+    arrow_tails_y_indices = Y * arrow_stride
+
+    # Determine the offsets in the image grid's X and Y directions
+
+    if quiver_projection_params is None:
+        # ROFF, RIFG, RUNW
+        # For L1 (ROFF, etc), `az_offsets_at_arrow_tails` indicates how many
+        # meters to offset in the image grid's y-direction (aka along-track).
+        # (Similar reasoning for slant range offsets and the x-direction.)
+        arrow_heads_y_offsets = az_offsets_at_arrow_tails
+        arrow_heads_x_offsets = rg_offsets_at_arrow_tails
+    else:
+        # GOFF, GUNW
+        # For L2 (GOFF, etc), `az_offsets_at_arrow_tails` indicates how many
+        # meters to offset in the satellite's along-track direction, which
+        # is NOT the same as the image grid's y-direction. So, do conversion.
+        if geo_grid is None:
+            raise ValueError(
+                f"`quiver_projection_params` was not None, but {geo_grid=};"
+                " it cannot be None."
+            )
```

Comment:
We can remove this check -- with the suggested changes above, `geo_grid` becomes a required parameter (named `coord_grid`).

```suggestion
```